### PR TITLE
Aborac/001

### PR DESCRIPTION
**Pull request by andrei-at-luther**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -430,17 +430,18 @@ func (c *Client) Kill() {
...
-			// If there is no error, then we attempt to wait for a graceful
-			// exit. If there was an error, we assume that graceful cleanup
-			// won't happen and just force kill.
-			graceful = err == nil
-			if err != nil {
-				// If there was an error just log it. We're going to force
-				// kill in a moment anyways.
-				c.logger.Warn("error closing client during Kill", "err", err)
-			}
+			client.Close()
```
Comment (bmatsuo-at-luthersystems) [inline client.go:433]: Do you maybe still want to log a warning if client.Close() returns one, like in the original code? You are still waiting for it to return after all. 🤷

File: rpc_client.go [hunk 2]
```diff
@@ -119,19 +119,11 @@ func (c *RPCClient) Close() error {
...
-	if err := c.stdout.Close(); err != nil {
-		return err
-	}
-	if err := c.stderr.Close(); err != nil {
-		return err
-	}
-	if err := c.broker.Close(); err != nil {
-		return err
-	}
+	// Close the other streams we have, but don't error check anything
```
Comment (bmatsuo-at-luthersystems) [inline rpc_client.go:122]: I guess a similar question here about whether these things can/should have any errors logged at a warning level. Maybe here you may don't have access to a good logger (I'm not familiar with the code yet).

**Review by bmatsuo-at-luthersystems — COMMENTED** [review]
I may be lacking some context here but I guess the logic seems reasonable. Just wondering of any of the ignored errors should be logged for good measure.

File: client.go (hunk 1 above)
Comment (andrei-at-luther) [reply client.go:433]: I'd say no to both; the whole point of the PR is to get rid of log messages that seem to show errors when in fact the occurrence of that error is perfectly normal and expected given the design.

Comment (bmatsuo-at-luthersystems) [reply client.go:433]: Oh yes. Of course. I guess when we were seeing this (at least in the tests I had done) It seems I generally a pair of log lines,

> 2021-04-01T09:13:49.059-0700 [WARN]  plugin: error closing client during Kill: err="unexpected EOF"
> 2021-04-01T09:13:49.059-0700 [WARN]  plugin: plugin failed to exit gracefully

Now, looking at this code, I suppose even after the changes here the test I ran would still print the second line. Is that how it should be? If so, maybe this client.Close() error could be stored and included as part of the "plugin failed to exit gracefully" log line if things do reach that point?

Comment (andrei-at-luther) [reply client.go:433]: The second line will still be printed if the plugin actually failed to exit gracefully within 2 seconds. That is unrelated to the error from `Close()`, which is a red herring (a graceful exit can occur even with such an error).

**Review by bmatsuo-at-luthersystems — APPROVED** [review]

**Pull request merged**